apps/aehttp/priv/swagger.json

```diff
@@ -59,31 +59,64 @@
           "200" : {
             "description" : "successful operation",
             "schema" : {
-              "$ref" : "#/definitions/Header"
+              "$ref" : "#/definitions/Top"
             }
           }
         },
         "security" : [ ]
       }
     },
-    "/block" : {
+    "/block-by-height" : {
       "get" : {
         "tags" : [ "external" ],
-        "description" : "Get the top block",
-        "operationId" : "GetBlock",
+        "description" : "Get a block by height",
+        "operationId" : "GetBlockByHeight",
         "consumes" : [ "application/json" ],
         "produces" : [ "application/json" ],
-        "parameters" : [ ],
+        "parameters" : [ {
+          "name" : "height",
+          "in" : "query",
+          "description" : "Height of the block to fetch",
+          "required" : true,
+          "type" : "integer"
+        } ],
         "responses" : {
           "200" : {
-            "description" : "top block",
+            "description" : "The block being found",
             "schema" : {
               "$ref" : "#/definitions/Block"
             }
           }
         },
         "security" : [ ]
-      },
+      }
+    },
+    "/block-by-hash" : {
```

Comment:
This is probably is subjective, but it doesn't look right to have block-by-hash and block-by-height. block/height/XX and block/hash/YY - also not perfect. block/ZZ and header for a type probably the cleanest.

Not a blocker but keep in mind when desigining the whole API

Reply:
I didn't like it either; got inspiration from testnet